lms/product/canvas/_plugin/misc.py

```diff
@@ -8,6 +8,34 @@
 
 
 class CanvasMiscPlugin(MiscPlugin):
+    def post_launch_assignment_hook(self, request, js_config, assignment):
```

Comment:
We use a similar `hook` approach for other differences.

The rationale here is that this might no be the perfect abstraction but helps to keep `if is_canvas` or `if is_d2l` out of the main flow. 

Once we have differences around the same thing in different LMSes we try for a better abstraction.